pkg/agent/types/annotations.go

```diff
@@ -17,7 +17,8 @@ package types
 const (
 	// NodeMACAddressAnnotationKey represents the key of the Node's MAC address in the Annotations of the Node.
 	NodeMACAddressAnnotationKey string = "node.antrea.io/mac-address"
-
 	// NodeTransportAddressAnnotationKey represents the key of the interface's IP address on which the Node transfers Pod traffic in the Annotations of the Node.
 	NodeTransportAddressAnnotationKey string = "node.antrea.io/transport-address"
+	// NodeWireGuardPublicKey represents the key of the Node's WireGuard public key in the Annotations of the Node.
+	NodeWireGuardPublicKey string = "node.antrea.io/wireguard-publickey"
```

Comment:
In somewhere, could you add some comments to explain the Node's public key is generated by antrea-agent and saved in the Node annotation?

Is it a good practice to save public key in annotation? @antoninbas @tnqn

Reply:
I have added a comment to the Init function of the WireGuard interface.
https://github.com/antrea-io/antrea/pull/2297/files#diff-f439c8b2008a71403262e863caf6300d6ac01117ef6714068140af934d08ad1eR23

Reply:
> Is it a good practice to save public key in annotation? @antoninbas @tnqn

@jianjuns I'm fine with it if exposing public key is not a security problem (is it?). I couldn't figure out a better way. If using secrets, we will have to deal with the cleanup, grant all secret to agent if having a secret per node, or deal with the update conflict if sharing a secret among all nodes.

Reply:
I do not see a security issue, but just wonder if any solution doing that. I am ok to go with the approach for now.

Reply:
I suggest to name it "wireguard-public-key" to follow the naming style strictly.
I can see all K8s annotation keys separate words:
```
      node-role.kubernetes.io/control-plane: ""
      node-role.kubernetes.io/master: ""
      node.kubernetes.io/exclude-from-external-load-balancers: ""
```

Reply:
updated.